wordpress/user.py

```diff
@@ -0,0 +1,41 @@
+import wordpress
+from wordpress.client import APIClient
+
+
+class WPUser:
+    email = None
+    roles = None
+    user_id = None
+    
+    def __init__(self, client=APIClient(), logger=wordpress.logger):
+        self.logger = logger
+        self.roles = set()
+        self.client = client
+
+
+    def fetch(self, email):
+        response = self.client.get_request(f'users/?search={email}')
+        body = response.json()
+        if len(body):
+            user_id = body[0]['id']
+            response = self.client.post_request(f'users/{user_id}')
+            body = response.json()
+            email_profile = body['email']
+            if email == email_profile:
+                self._convert_to_resource(response)
+
+        return self
+
+    def _convert_to_resource(self, response):
+        body = response.json()
+        self.roles = body['roles']
+        self.username = body['username']
+        self.email = body['email']
+        self.user_id = body['id']
+
+    def update_as_partner(self):
```

Comment:
I would rather keep this user-resource class implement CRUD only. Implement a set of fetch, update, create, delete methods is what we need here. Then, this more use-case-specific logic can go into the Celery task itself. Only if we see the need to support more advanced methods beyond simple CRUD, I would consider adding them here in the future.

TL;DR: I turn this into (disclaimer, this is pseudo-codish):

```py
def update(self, **kwargs):
  response = self.client.post_request(f'users/{self.user_id}', data=locals())
  self._convert_to_resource(response)
```

and then on `update_user_as_partner`, which is at one level of abstraction higher:

```py
def update_user_as_partner(email):
  wp_user = WPUser().fetch(email)
  if wordpress.wp_partner_role not in self.roles:
    new_roles = ",".join(self.roles + wordpress.wp_partner_role)
    wp_user.update(roles=new_roles)
```
:ok_hand:

Reply:
Looks good!